grails-app/assets/javascripts/document.js

```diff
@@ -61,13 +82,30 @@ function DocumentViewModel (doc, owner, settings) {
         return (self.filename() ? "Change file" : "Attach file");
     });
 
+    self.transients = {};
+    self.transients.dateCreated = new Date();
```

Comment:
Here `dateCreated` and `lastUpdated` is of type `Date` and below it is of type `String`. It should be of type `String` here as well?